cloud/deploy/bundle.go

```diff
@@ -0,0 +1,247 @@
+	if input.Wait {
+		err = deployment.HealthPoll(currentDeployment.Id, currentDeployment.WorkspaceId, dagOnlyDeploySleepTime, tickNum, timeoutNum, platformCoreClient)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func uploadBundle(tarDirPath, bundlePath, uploadURL string, prependBaseDir bool) (string, error) {
+	tarFilePath := filepath.Join(tarDirPath, "bundle.tar")
+	defer func() {
+		err := os.Remove(tarFilePath)
+		if err != nil {
+			fmt.Println("\nFailed to delete tar file: ", err.Error())
+			fmt.Println("\nPlease delete the tar file manually from path: " + tarFilePath)
+		}
+	}()
+
+	// Generate the bundle tar
+	err := fileutil.Tar(bundlePath, tarFilePath, prependBaseDir)
+	if err != nil {
+		return "", err
+	}
+
+	tarFile, err := os.Open(tarFilePath)
+	if err != nil {
+		return "", err
+	}
+	defer tarFile.Close()
+
+	versionID, err := azureUploader(uploadURL, tarFile)
+	if err != nil {
+		return "", err
+	}
+
+	return versionID, nil
+}
+
+func getDeployment(organizationID, deploymentID string, platformCoreClient astroplatformcore.CoreClient) (*astroplatformcore.Deployment, error) {
+	resp, err := platformCoreClient.GetDeploymentWithResponse(context.Background(), organizationID, deploymentID)
+	if err != nil {
+		return nil, err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.JSON200, nil
+}
+
+func createBundleDeploy(organizationID string, input *DeployBundleInput, deployGit *astrocore.DeployGit, coreClient astrocore.CoreClient) (*astrocore.Deploy, error) {
+	request := astrocore.CreateDeployRequest{
+		Description:     &input.Description,
+		Type:            astrocore.CreateDeployRequestTypeBUNDLE,
+		BundleMountPath: &input.MountPath,
+		BundleType:      &input.BundleType,
+	}
+	if deployGit != nil {
+		request.Git = &astrocore.CreateDeployGitRequest{
+			Provider:   astrocore.CreateDeployGitRequestProvider(deployGit.Provider),
+			Repo:       deployGit.Repo,
+			Account:    deployGit.Account,
+			Path:       deployGit.Path,
+			Branch:     deployGit.Branch,
+			CommitSha:  deployGit.CommitSha,
+			CommitUrl:  fmt.Sprintf("https://github.com/%s/%s/commit/%s", deployGit.Account, deployGit.Repo, deployGit.CommitSha),
+			AuthorName: deployGit.AuthorName,
+		}
+	}
+	resp, err := coreClient.CreateDeployWithResponse(context.Background(), organizationID, input.DeploymentID, request)
+	if err != nil {
+		return nil, err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return resp.JSON200, nil
+}
+
+func finalizeBundleDeploy(organizationID, deployID, tarballVersion string, input *DeployBundleInput, coreClient astrocore.CoreClient) (*astrocore.UpdateDeployResponse, error) {
+	request := astrocore.UpdateDeployRequest{
+		BundleTarballVersion: &tarballVersion,
+	}
+	resp, err := coreClient.UpdateDeployWithResponse(context.Background(), organizationID, input.DeploymentID, deployID, request)
```

Comment:
`finalizeDeploy` calls `platformCoreClient.FinalizeDeployWithResponse` -- what's the difference bewtween finalizeDeploy and updateDeploy? (and why do we have both)

Reply:
In v1alpha1 updateDeploy doesn't allow the user to just update the deploy entry, it is only used to tell Astro to finalize the deploy (i.e. roll out the image/DAGs to the deployment). In v1beta1 we decided to follow CRUD semantics more closely and have updateDeploy be for updating the deploy entry and finalizeDeploy be a verb route (/finalize) that instructs Astro to roll out the deploy.

Reply:
+1 to what jeremy said. We will eventually get the v1alpha1 and v1beta1 terminology the same, once we know all users have moved to a cli version that uses v1beta1 endpoints and we have a couple of customers using v1alpha1 endpoints, those will be migrated to using v1beta1 endpoints